gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java

```diff
@@ -923,6 +923,47 @@ public SubscriptionEntity fail(String subscriptionId, String failureCause) {
         }
     }
 
+    @Override
+    public SubscriptionEntity notifyError(String subscriptionId, String failureCause) {
+        try {
+            logger.debug("Failed subscription Notification{}", subscriptionId);
+            Subscription subscription = subscriptionRepository
+                .findById(subscriptionId)
+                .orElseThrow(() -> new SubscriptionNotFoundException(subscriptionId));
+            final Date now = new Date();
+            subscription.setUpdatedAt(now);
+            subscription.setConsumerPausedAt(null);
+            subscription.setFailureCause(failureCause);
+            subscription.setReason(failureCause);
```

Comment:
we should not update reason because it is used by the publisher to explain why we refused a subscription during the approval process